ETL/Core/Expression.cs

```diff
@@ -90,8 +90,54 @@ public static bool AddValuesToDatatableDestination(DataTable source, DataTable d
                             {
                                 continue;
                             }
-                            var value = GetValue(expRow, row, mapDt);
-                            newRow[col] = value;
+                            if (col.DataType == System.Type.GetType("System.Int32"))
+                            {
+                                int value;
+                                Int32.TryParse(GetValue(expRow, row, mapDt), out value);
+                                newRow[col] = value;
+                            }
+                            else if (col.DataType == System.Type.GetType("System.DateTime"))
+                            {
+                                DateTime value;
+                                DateTime.TryParse(GetValue(expRow, row, mapDt), out value);
+                                newRow[col] = value;
+                            }
+                            else if (col.DataType == System.Type.GetType("System.String"))
+                            {
+                                string value = GetValue(expRow, row, mapDt).ToString() ;
+                                newRow[col] = value;
+                            }
+                            else if (col.DataType == System.Type.GetType("System.Int16"))
+                            {
+                                short value;
+                                Int16.TryParse(GetValue(expRow, row, mapDt), out value);
+                                newRow[col] = value;
+                            }
+                            else if (col.DataType == System.Type.GetType("System.Boolean"))
+                            {
+                                bool value;
+                                Boolean.TryParse(GetValue(expRow, row, mapDt), out value);
+                                newRow[col] = value;
+                            }
+                            else if (col.DataType == System.Type.GetType("System.TimeSpan"))
+                            {
+                                TimeSpan value;
+                                TimeSpan.TryParse(GetValue(expRow, row, mapDt), out value);
+                                newRow[col] = value;
+                            }
+                            else if (col.DataType == System.Type.GetType("System.Double"))
+                            {
+                                double value;
+                                Double.TryParse(GetValue(expRow, row, mapDt), out value);
+                                newRow[col] = value;
+                            }
+                            else if (col.DataType == System.Type.GetType("System.Char"))
+                            {
+                                char value;
+                                Char.TryParse(GetValue(expRow, row, mapDt), out value);
+                                newRow[col] = value;
+                            }
```

Comment:
else what happens? 
let's add an else condition and have the value as a string in it